src/data/markdown/docs/20 jslib/01 jslib/05 aws/01 S3Client.md

```diff
@@ -0,0 +1,80 @@
+---
+title: 'S3Client'
+head_title: 'S3Client'
+description: 'S3Client allows interacting with AWS S3 buckets and objects'
+excerpt: 'S3Client class allows interacting with AWS S3 buckets and objects'
+---
+
+S3Client allows interacting with AWS S3's buckets and objects. Namely, it allows the user to list buckets and the objects they contain, as well as download, uploading, and deleting objects. The S3Client operations are blocking, and we recommend reserving their usage to the [init context](/javascript-api/init-context/), or to the [`setup`](/using-k6/test-life-cycle/) and [`teardown`](/using-k6/test-life-cycle/) functions.
+
+S3Client is included in both the dedicated jslib `s3.js` bundle, and the `aws.js` one, containing all the services clients.
+
+### Methods
+
+| Function                                                                                                                  | Description                                           |
+| :------------------------------------------------------------------------------------------------------------------------ | :---------------------------------------------------- |
+| [listBuckets()](/javascript-api/jslib/aws/s3client/s3client-listbuckets)                                                  | List the buckets the authenticated user has access to |
+| [listObjects(bucketName, [prefix])](/javascript-api/jslib/aws/s3client/s3client-listobjects-bucketname-prefix)            | List the objects contained in a bucket                |
+| [getObject(bucketName, objectKey)](/javascript-api/jslib/aws/s3client/s3client-getobject-bucketname-objectkey)            | Download an object from a bucket                      |
+| [putObject(bucketName, objectKey, data)](/javascript-api/jslib/aws/s3client/s3client-putobject-bucketname-objectkey-data) | Upload an object to a bucket                          |
+| [deleteObject(bucketName, objectKey)](/javascript-api/jslib/aws/s3client/s3client-deleteobject-bucketname-objectkey)      | Delete an object from a bucket                        |
+
+### Throws
+
+S3 Client methods will throw errors in case of failure.
+
+| Error                 | Condition                                                  |
+| :-------------------- | :--------------------------------------------------------- |
+| InvalidSignatureError | when invalid credentials were provided.                    |
+| S3ServiceError        | when AWS replied to the requested operation with an error. |
+
+### Example
```

Comment:
If we are recommending users to call these APIs in `setup()` - perhaps we should provide an example like that?

Reply:
Good call. I modified both the `S3Client` and `SecretsManagerClient` main examples to be more real-world oriented and exploit `setup`, `teardown` and `handleSummary` indeed 👍🏻